### PR TITLE
feat: transaction history page and export

### LINKED ISSUE #1006: View and Download Tx History
**note - this is reserved for @stackedq - if you start work on this in parallel your PR will be closed**

### Overview

Allow users to download their Tx history within ShapeShift as a .csv for easy import into crypto tax applications.

Users will be able to navigate to specific "Transaction History" button in the global navigation, as well as see specific asset transaction history from within the "Asset" page.

1. User sees icon in the global navigation that links to ‘Trade History’ section
2. Transaction History Section contains:
    a. Search by - asset, chain
    b. Filter by date range, sends, receives, trades, asset type - Following established styles for web v2. Trade rows should be expanding drawers that include the same information currently seen on web v2. Note: infinite scroll here.
    c. Download CSV link. Note: Download should download a filtered list if user has selected filters.

3. In asset selection page, user can select a single asset and press “View Transaction History” (see UX Screen 3) and will see a filtered list of transactions with that asset4. 

CSV is below. Exported from beta.shapeshift.com so formatting can stay the same.

All headers are listed below

id    status    timestamp    rate    minerFee    inputAmount    inputCurrency    inputAddress    inputTxid    outputAmount    outputCurrency    outputAddress    outputTxid    refundAddress    refundCurrency    refundTxid

[example_trade_history.csv](https://github.com/shapeshift/web/files/8164996/example_trade_history.csv)

### Reference

Placeholder for engineering.

### Acceptance Criteria

Transaction history button is included in global navigation and all transactions for connected wallet are shown - Including date and time, originating asset, destination asset, fee, and link to block explorer.

Link to download csv for desired/filtered data functions properly.

Transaction history is included on each individual asset page, properly sorted by that specific asset.

### Screenshots/Mockups

UX:
Transaction History Overview 
txhistory

Transaction History with expanded details
Screen Shot 2022-03-09 at 11 10 31 AM

Transaction History with expanded filters. NOTE: User should also be able to filter and search by asset type (ETH, USDC, etc for all supported coins) within the filter section.
filter

Transaction history within specific asset page. 
txhistorysmall

Mobile Mockups

Tx history view
Mobile

Expanded Tx history view
Mobile_Expand

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info

### PR DESCRIPTION
**Pull request by stackedq**
## Description

Implementing transaction history page
WIP:
  - ~~Search & filter functionality~~
  - ~~Download csv functionality~~
  - ~~Showing USD amount for each cell in `TransactionRow` (currently it shows transaction time)~~
  - Showing `order route` correct value in `TransactionDetails` -> I don't know if I did it correctly, so please give me feedback.
  - ~~Link from `Asset` page~~
  - ~~Mobile/tablet changes~~
  - ~~waiting on date picker design~~
  - ~~waiting on this [PR](https://github.com/shapeshift/web/pull/1124)~~

 
Things to be decided:
 - ~~Chakra-UI does not include a `DatePicker` component, there are multiple options out there that could be integrated. currently, the browser default `date` is being used which is unacceptable. added `react-datepicker`, but waiting on the design from the product team~~ -> design is ready
 - ~~Fiat amounts shown in `TransactionRow` are using current market data price, which generates false information, I need help/suggestions on how to get the fiat rate at transaction `blockTime`~~ -> skipped for now

> Note: contains multiple `console.info` for developing purpose, will be deleted before marking PR for Review.

> Note: currently I created a new directory called TransactionHistoryRows, this is intended to lower the conflicts, once this PR is completed, the directory will be renamed to Transactions and the TransactionRow component will get extra props to work both in recent transactions section of asset/account pages and in transaction history page, and asset/account page transaction tab.

> Files structure could be improved, but I tried not to overthink it, so waiting for suggestions.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #1006

## Screenshots (if applicable)

filters open and datepicker
filters
480px-breakpoint

768px-breakpoint

992px-breakpoint

1280px-breakpoint and above

light-mode
light-mode
light-mode with filters and datepicker
light-mode-filters

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
https://github.com/shapeshift/web/pull/1124 is in @stackedq 👌

**Comment by stackedq** [comment]
Marking as draft due to arrival of the date picker design

File: src/pages/TransactionHistory/components/FilterGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        <Text translation={title} />
+      </Button>
+      <Collapse in={isOpen} unmountOnExit>
+        <Box px={2} mb={2}>
+          <GroupComponent value={value || []} onChange={onChange} name={name}>
+            {options.map(([title, optionValue, CustomComponent]: Option) => (
+              <>
+                <Box key={optionValue} py={1}>
+                  <InputComponent value={optionValue}>
+                    <Text translation={title} fontWeight='300' />
```
Comment (reallybeard) [inline src/pages/TransactionHistory/components/FilterGroup.tsx:68]: Let's leave the weights to the default 500 or whatever it is here. 300 looks a little too thin to me.

File: src/pages/TransactionHistory/components/FilterGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  } = useController({ control, name })
+  const GroupComponent = allowMultipleOptions ? CheckboxGroup : RadioGroup
+  const InputComponent = allowMultipleOptions ? Checkbox : Radio
+  return (
+    <Box px={2}>
+      <Button
+        justifyContent='space-between'
+        alignItems='center'
+        variant='ghost'
+        fontWeight='400'
```
Comment (reallybeard) [inline src/pages/TransactionHistory/components/FilterGroup.tsx:45]: Same here, can leave it at the same weight.

File: src/pages/TransactionHistory/components/FilterGroup.tsx (hunk 1 above)
Comment (stackedq) [reply src/pages/TransactionHistory/components/FilterGroup.tsx]: done

File: src/pages/TransactionHistory/components/FilterGroup.tsx (hunk 2 above)
Comment (stackedq) [reply src/pages/TransactionHistory/components/FilterGroup.tsx]: done

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
lots of comments, please use your discretion as to what's actually required (the 18s) and memoizing selectors, vs comments or stylistic stuff

File: src/pages/TransactionHistory/components/DatePicker/DatePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        <FaCalendarAlt />
+      </InputLeftElement>
+      <Input
+        as={ReactDatePicker}
+        selected={value || ''}
+        onChange={onChange}
+        bg={useColorModeValue('gray.300', 'gray.750')}
+        color={useColorModeValue('blue.300', 'blue.200')}
+        name={name}
+        placeholderText='00/00/0000'
```
[the remark this answers is not in the document]
Comment (stackedq) [reply src/pages/TransactionHistory/components/DatePicker/DatePicker.tsx:25]: Just the placeholder? for both of the inputs?

File: src/components/TransactionHistoryRows/TransactionDate.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import dayjs from 'dayjs'
+import { RawText } from 'components/Text'
+
+export const TransactionDate = ({ blockTime }: { blockTime: number }) => (
+  <RawText fontWeight='bold' lineHeight='1' mb={6} whiteSpace='nowrap'>
+    {dayjs(blockTime * 1000).format('MMMM DD, YYYY')}
```
[the remark this answers is not in the document]
Comment (stackedq) [reply src/components/TransactionHistoryRows/TransactionDate.tsx:6]: Should I add a TODO comment in the code for future?

**Review by 0xdef1cafe — APPROVED** [review]

**Review by asamere — APPROVED** [review]
🍭

**Pull request merged**